### PR TITLE
feat: preparations of covector/action for building and publishing preview packages

### PR DESCRIPTION
**Pull request by minkimcello**
## Motivation
Initial efforts for #147. We want to be able to run covector in github actions to publish preview packages.

## Approach
- Action's `with.command` property (for the workflow) is now required and I removed the `default: "status"` in `package/action/action.yml`.
  - The required boolean doesn't seem to actually make a difference. You can run the action either locally (`./packages/action/`) or by calling it from the repository (`jbolda/covector/packages/action`) and it won't throw an error even if you don't provide a command. 🤷‍♂️
    - And this is the reason why we need to handle `if no command = throw error` inside the action manually.
- Specified the `action/dist/*` directory and lock files as binary in `.gitattributes` so that it doesn't get counted as contributions (but still gets included in the repository (a little trick I learned from Robbie (but I think it takes effect after it's merged?))).
- Took out the default "status" command _within_ the action itself. So now when it's given no-command or an unknown command, it'll thrown an error.
- Added a preview workflow for the covector repository.
- If there is a "preview" label on the PR, the action will run `covector preview`, if not, it'll just console log out to say that it's not going to try and publish anything because there is no label on the PR.
  - Added a placeholder console log for `covector preview` (to scope this work to a separate PR).

## TODOs
- [x] I hard coded the accepted label as "preview". Should this be configurable? Added in #187 
- [x] ~Are we doing "previews" for Github releases (by doing createRelease but with "pre-release")?~ No
- [x] Do we want the action to generate a comment on the pull request like the current frontside preview action? Will tackle in #187
- [x] Need to add change file for action package

### CODE REVIEW

**Comment by jbolda** [comment]
> Took out the default "status" command within the action itself. So now when it's given no-command or an unknown command, it'll thrown an error.

I presume an error with a "help" output listing all of the commands? Sounds good to me.

> Are we doing "previews" for Github releases (by doing createRelease but with "pre-release")?

I don't think we do for previews, that will become really noisy and it relies on a git tag mostly off of the default branch. I don't see a strong value proposition in it, and things will get funky.

> Do we want the action to generate a comment on the pull request like the current frontside preview action?

I think that makes sense to me. Do you have any thoughts with the comment as you have used the previews? Thinking a verbatim implementation?

**Comment by minkimcello** [comment]
@jbolda 

> I think that makes sense to me. Do you have any thoughts with the comment as you have used the previews? Thinking a verbatim implementation?

Since we're doing this in Javascript, generating the comment via octokit will be pretty straightforward. 👌

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "watch": "tsc --build --watch tsconfig.packages.json",
     "clean": "tsc --build --clean tsconfig.packages.json",
     "wipe": "npm run clean -ws && rimraf node_modules dist tsconfig.tsbuildinfo && npm i",
+    "build:action": "yarn workspace action pkg",
```
Comment (jbolda) [inline package.json:14]: I could be convinced either way that we need this at the root, but we should use npm for it.
```suggestion
    "build:action": "npm run pkg -w action",
```

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,3 @@
+packages/action/dist/* binary
+**/yarn.lock binary
```
Comment (jbolda) [inline .gitattributes:2]: Please remove the `yarn.lock`.

File: packages/action/src/index.ts [hunk 3]
```diff
@@ -98,7 +104,50 @@ export function* run(): Generator<any, any, any> {
...
         return covectored;
       }
-    }
+    } else if (command === "preview") {
+      const previewLabel = github?.context?.payload?.pull_request?.labels?.filter(({ name } : { name: String }) => name === 'preview').length;
+      if (!previewLabel) {
+        console.log('Not publishing any preview packages because the "preview" label has not been applied to this pull request.')
+      } else {
+        let covectored: Covector;
+        // if (core.getInput("createRelease") === "true" && token) {
```
Comment (jbolda) [inline packages/action/src/index.ts:113]: As we discussed, we are going to deal with the commented-out code in #187.

**Review by jbolda — APPROVED** [review]
Excited to see this coming down the pipeline!

Can we remove the `yarn.lock`? It should be built with npm@7 and use the lock file that is in the root.

**Pull request merged**